packages/typography/src/elements/CodeBlock.tsx

```diff
@@ -59,47 +93,54 @@ export const CodeBlock = React.forwardRef<HTMLPreElement, ICodeBlockProps>(
     };
 
     return (
-      <StyledCodeBlockContainer {...containerProps} ref={containerRef} tabIndex={containerTabIndex}>
-        <Highlight
-          Prism={Prism}
-          code={code ? code.trim() : ''}
-          language={LANGUAGES.includes(language as Language) ? (language as Language) : 'tsx'}
+      isPrismImported && (
+        <StyledCodeBlockContainer
+          {...containerProps}
+          ref={containerRef}
+          tabIndex={containerTabIndex}
         >
-          {({ className, tokens, getLineProps, getTokenProps }) => (
-            <StyledCodeBlock className={className} ref={ref} isLight={isLight} {...other}>
-              {tokens.map((line, index) => (
-                <StyledCodeBlockLine
-                  {...getLineProps({ line })}
-                  key={index}
-                  language={language}
-                  isHighlighted={highlightLines && highlightLines.includes(index + 1)}
-                  isLight={isLight}
-                  isNumbered={isNumbered}
-                  diff={getDiff(line)}
-                  size={size}
-                >
-                  {line.map((token, tokenKey) => (
-                    <StyledCodeBlockToken
-                      {...getTokenProps({ token })}
-                      key={tokenKey}
-                      isLight={isLight}
+          <Highlight
+            code={code ? code.trim() : ''}
+            language={LANGUAGES.includes(language!) ? (language as Language) : 'tsx'}
+          >
+            {({ className, tokens, getLineProps, getTokenProps }) => (
+              <ThemeProvider
+                theme={parentTheme => ({
+                  ...parentTheme,
+                  colors: { ...parentTheme.colors, base: isLight ? 'light' : 'dark' }
+                })}
+              >
```

Comment:
Using the coerced mode theming technique here to render styled components based on the `isLight` prop.